frontend/src/components/common/StatusFilter.tsx

```diff
@@ -0,0 +1,138 @@
+import { useTypedSelector } from '../../redux/reducers/reducers';
+
+export interface PureStatusesAutocompleteProps {
+  statusNames: string[];
+  onChange: (event: React.ChangeEvent<{}>, newValue: string[]) => void;
+  filter: { statuses: Set<string>; search: string };
+}
+
+export function PureStatusesAutocomplete({
+  statusNames,
+  onChange: onChangeFromProps,
+  filter,
+}: PureStatusesAutocompleteProps) {
+  const theme = useTheme();
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [statusInput, setStatusInput] = React.useState<string>('');
+  const maxStatusesChars = 12;
+
+  const onInputChange = (event: object, value: string, reason: string) => {
+    // For some reason, the AutoComplete component resets the text after a short
+    // delay, so we need to avoid that or the user won't be able to edit/use what they type.
+    if (reason !== 'reset') {
+      setStatusInput(value);
+    }
+  };
+
+  const onChange = (event: React.ChangeEvent<{}>, newValue: string[]) => {
+    // Now we reset the input so it won't show next to the selected namespaces.
+    setStatusInput('');
+    onChangeFromProps(event, newValue);
+  };
+
+  return (
+    <Autocomplete
+      multiple
+      id="statuses-filter"
+      autoComplete
+      options={statusNames}
+      onChange={onChange}
+      onInputChange={onInputChange}
+      inputValue={statusInput}
+      // We reverse the namespaces so the last chosen appear as the first in the label. This
+      // is useful since the label is ellipsized and this we get to see it change.
+      value={[...filter.statuses.values()].reverse()}
+      renderOption={(props, option, { selected }) => (
+        <li {...props}>
+          <Checkbox
+            icon={<Icon icon="mdi:checkbox-blank-outline" />}
+            checkedIcon={<Icon icon="mdi:check-box-outline" />}
+            style={{
+              color: selected ? theme.palette.primary.main : theme.palette.text.primary,
+            }}
+            checked={selected}
+          />
+          {option}
+        </li>
+      )}
+      renderTags={(tags: string[]) => {
+        if (tags.length === 0) {
+          return <Typography variant="body2">{t('translation|All statuses')}</Typography>;
+        }
+
+        let statusesToShow = tags[0];
+        const joiner = ', ';
+        const joinerLength = joiner.length;
+        let joinnedStatuses = 1;
+
+        tags.slice(1).forEach(tag => {
+          if (statusesToShow.length + tag.length + joinerLength <= maxStatusesChars) {
+            statusesToShow += joiner + tag;
+            joinnedStatuses++;
+          }
+        });
+
+        return (
+          <Typography style={{ overflowWrap: 'anywhere' }}>
+            {statusesToShow.length > maxStatusesChars
+              ? statusesToShow.slice(0, maxStatusesChars) + '…'
+              : statusesToShow}
+            {tags.length > joinnedStatuses && (
+              <>
+                <span>,&nbsp;</span>
```

Comment:
If at all possible, we avoid html tags and use Mui components. Could you please switch these ones out for mui components?